### PR TITLE
nixos/pfix-srsd: add module

### PR DESCRIPTION
**Pull request by jerith666**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/mail/pfix-srsd.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        default = false;
+        description = "Whether to run the postfix sender rewriting scheme daemon.";
+      };
+
+      domain = mkOption {
+        description = "The domain for which to enable srs";
+      };
+
+      secret = mkOption {
+        description = "The secret data used to encode the SRS address";
```
Comment (joachifm) [inline nixos/modules/services/mail/pfix-srsd.nix]: How important is it that this remains secret? If at all, consider not copying it into the store.

File: nixos/modules/services/mail/pfix-srsd.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  options = {
+
+    services.pfix-srsd = {
+      enable = mkOption {
+        default = false;
+        description = "Whether to run the postfix sender rewriting scheme daemon.";
+      };
+
+      domain = mkOption {
```
Comment (joachifm) [inline nixos/modules/services/mail/pfix-srsd.nix]: Please declare types for the options.

File: nixos/modules/services/mail/pfix-srsd.nix (hunk 1 above)
Comment (jerith666) [reply nixos/modules/services/mail/pfix-srsd.nix]: I think it is rather important -- possession of this secret means you can forge bounces through this mail server.  Looking into duplicating the approach used by `postsrsd.nix`.

**Comment by jerith666** [comment]
Okay, secrets moved out of store and types added.  Thanks for the review!

**Pull request merged**